src/mds/CInode.cc

```diff
@@ -5410,7 +5410,13 @@ void CInode::maybe_ephemeral_rand(bool fresh)
     return;
   }
 
-  double threshold = get_ephemeral_rand();
+  /* not precomputed? */
+  if (threshold < 0.0) {
+    threshold = get_ephemeral_rand();
+  }
+  if (threshold <= 0.0) {
```

Comment:
Shouldn't it be:
```
else if (threshold == 0) {
 return;
}
```
Had it been pre-computed, it need not hit the second if statement again

Reply:
Please disregard the previous comment. I wasn't thinking straight.